DNSMADEASY: staticcheck errors

I'm moving from go-lint to staticcheck (https://staticcheck.io/). It finds some issues that I think are worth fixing.

```
providers/dnsmadeeasy/restApi.go:39:6: type apiEmptyResponse is unused (U1000)
providers/dnsmadeeasy/restApi.go:101:13: unnecessary use of fmt.Sprintf (S1039)
providers/dnsmadeeasy/types.go:142:3: this value of err is never used (SA4006)
```

--- a/providers/dnsmadeeasy/restApi.go
+++ b/providers/dnsmadeeasy/restApi.go
@@ -95,7 +95,7 @@ func (restApi *dnsMadeEasyRestAPI) singleDomainCreate(data singleDomainRequestDa
 
 	req := &apiRequest{
 		method:   "POST",
-		endpoint: fmt.Sprintf("dns/managed/"),
+		endpoint: "dns/managed/",
 		data:     jsonData,
 	}
 

--- a/providers/dnsmadeeasy/restApi.go
+++ b/providers/dnsmadeeasy/restApi.go
@@ -36,9 +36,6 @@ type apiErrorResponse struct {
 	Error []string `json:"error"`
 }
 
-type apiEmptyResponse struct {
-}
-
 type apiRequest struct {
 	method   string
 	endpoint string

--- a/providers/dnsmadeeasy/types.go
+++ b/providers/dnsmadeeasy/types.go
@@ -135,8 +135,8 @@ func toRecordConfig(domain string, record *recordResponseDataEntry) *models.Reco
 	} else if record.Type == "SRV" {
 		err = rc.SetTargetSRV(uint16(record.Priority), uint16(record.Weight), uint16(record.Port), record.Value)
 	} else if record.Type == "CAA" {
-		value, err := strconv.Unquote(record.Value)
-		if err != nil {
+		value, unquoteErr := strconv.Unquote(record.Value)
+		if unquoteErr != nil {
 			panic(err)
 		}
 		err = rc.SetTargetCAA(uint8(record.IssuerCritical), record.CaaType, value)

